match/lib/match/importer.rb

```diff
@@ -125,13 +140,46 @@ def import_cert(params, cert_path: nil, p12_path: nil, profile_path: nil)
       storage.clear_changes if storage
     end
 
-    def ensure_valid_file_path(file_path, file_description, file_extension, optional: false)
+    def resolve_inputs(params, cert_path, p12_path, profile_path)
+      cert_path ||= params[:import_certificate_file_path]
```

Comment:
would it be better to modify `ensure_valid_file_path` and pass the params and key instead?

So the code would become something like

```
cert_path = ensure_valid_file_path(params, :import_certificate_file_path, "Certificate", ".cer", optional: true)
p12_path = ensure_valid_file_path(params, :import_certificate_private_key_file_path, "Private key", ".p12", optional: true)
profile_path = ensure_valid_file_path(params, :import_provisioning_profile_file_path,  "Provisioning profile", ".mobileprovision or .provisionprofile", optional: true)


[...]

def ensure_valid_file_path(params, file_path_param_key, file_description, file_extension, optional: false, skip_prompt: false)
  
  file_path = params[file_path_param_key]
  unless file_path
    file_path = UI.input("#{file_description} (#{file_extension}) path#{optional_file_message}:")
  end
```

and for the `import_suppress_ui_file_path_prompts` I wonder if we shouldn't aim for a generic suppress prompts instead.

Reply:
regarding handling of those params, possible, but feel like its splitting hairs at this point. 

regarding a more generic prompt suppression, sounds like a reasonable idea, but I'm not really equipped to tackle that everywhere at the moment timewise. Would really like to see these granular import flags rolled in as this is a part of a larger project I'm working on.